File: Topic-Wise/Recursion/C++/Removeduplicates.cpp
//
// Created by Siddhivinayak Dubey on 11/04/24.
//
#include "iostream"
using namespace std;
#include "unordered_set"

unordered_set<char> Chars;

void removeduplicates(string s, int idx){
    if(idx == s.length()){
        cout<<s;
        return;
    }
    if(Chars.find(s[idx]) == Chars.end()){
        Chars.insert(s[idx]);
    }
    else{
        s.erase(idx,1);
        idx-=1;
    }
    removeduplicates(s, idx+1);
}

int main(){
    removeduplicates("abbcdddebdceba", 0);
}

File: Topic-Wise/Recursion/Python/removeduplicates.py
Chars = set()
result =""
def removeduplicates(s, idx):
    # global result
    # if(idx == len(s)):
    #     print(result)
    #     return
    # if(s[idx] not in Chars):
    #     Chars.add(s[idx])
    #     result+=s[idx]
    # removeduplicates(s,idx+1)

    #OR
    string = set(s)
    str = ''
    for chars in string:
        str+=chars;
    print(str)

if __name__ == '__main__':
    removeduplicates("abbcdddebdceba", 0);
